.github/workflows/tests.yml

```diff
@@ -229,6 +232,16 @@ jobs:
 
       - uses: codecov/codecov-action@29386c70ef20e286228c72b668a06fd0e8399192
 
+      - name: Upload test results to BuildPulse for flaky test detection
+        if: '${{ matrix.buildpulse }} && !cancelled()' # Run this step even when the tests fail. Skip if the workflow is cancelled or BuildPulse isn't enabled for matrix job.
+        uses: Workshop64/buildpulse-action@main
+        with:
+          account: 2988
+          repository: 377629038
+          path: ./Library/Homebrew/test/junit
+          key: ${{ secrets.BUILDPULSE_ACCESS_KEY_ID }}
+          secret: ${{ secrets.BUILDPULSE_SECRET_ACCESS_KEY }}
```

Comment:
I guess this is the tricky bit as it wouldn't work for PRs from forks. Thoughts?

Reply:
That's right. Although now that the BuildPulse uploads have moved to the macOS job (which only runs on Homebrew/brew and doesn't run in forks), I think that resolves the issue. Does that sound right?

Note: For testing/demonstration purposes, this PR [enables the macOS jobs for all forks here](https://github.com/jasonrudolph/brew/pull/2/files#diff-1db27d93186e46d3b441ece35801b244db8ee144ff1405ca27a163bfe878957fL249), but that change won't be part of the upstream PR.